src/dns_internal.cc

```diff
@@ -282,6 +288,29 @@ static unsigned short idnsQueryID(void);
 static void idnsSendSlaveAAAAQuery(idns_query *q);
 static void idnsCallbackOnEarlyError(IDNSCB *callback, void *cbdata, const char *error);
 
+void
+idns_query::makeNameToLookup()
+{
+    strcpy(name, orig); // default is always hostname-only.
+
+    // try to find the next suitable suffix to append
+    const auto prefixLen = strlen(orig) + 1 /* the joiner '.' */;
+    while (domain < npc) {
+        const auto need = prefixLen + strlen(searchpath[domain].domain);
+        if (need > NS_MAXDNAME) {
+            debugs(23, 3, "searchpath FQDN for '" << orig << "." << searchpath[domain].domain << "' too long. skip.");
+            ++domain;
+            continue; // skip this searchpath
+        }
+        strcat(name, ".");
+        strcat(name, searchpath[domain].domain);
+        debugs(78, 3, "searchpath[" << domain << "] used to create " << name);
+        ++domain;
+        return; // try using this FQDN
+    }
+    // else use the default (orig only)
```

Comment:
There is no `if` for this "else", so I recommend dropping it:


```suggestion

    // use the default (orig only)
```

I do not insist on this change.